### PR TITLE
Implement job runs as relationship for jobs

### PR DESCRIPTION
**Pull request by vyzaldysanchez**
Closes: https://app.shortcut.com/chainlinklabs/story/20930/implement-runs-field-on-a-job

This aims to add a relationship to jobs so consumers can also get the pipeline runs related.

### CODE REVIEW

File: core/web/schema/type/job_run.graphql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+type JobRun {
+    outputs: [String]!
+    allErrors: [String]!
```
Comment (jkongie) [inline core/web/schema/type/job_run.graphql:3]: This could be [String!]! I believe

File: core/web/schema/type/job_run.graphql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    fatalErrors: [String]!
```
Comment (jkongie) [inline core/web/schema/type/job_run.graphql:4]: This could be [String!]! I believe

File: core/web/schema/type/job_run.graphql [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    inputs: String!
+    createdAt: Time!
+    finishedAt: Time
+    taskRuns: [TaskRun!]!
+    observationSource: String!
```
Comment (jkongie) [inline core/web/schema/type/job_run.graphql:9]: I don't think we need the observation source.

We could add a separate ticket which allows you to fetch the job associated with the run which will contain the observation source

**Review by jkongie — CHANGES_REQUESTED** [review]

File: core/web/schema/type/job_run.graphql (hunk 3 above)
Comment (vyzaldysanchez) [reply core/web/schema/type/job_run.graphql]: Sure, we could create a ticket to work on that

**Review by jkongie — CHANGES_REQUESTED** [review]
Add a task run resolver test. Because there is no query to fetch a job run or a task run individually, you can do this by requesting a job with a job run containing a task run and testing all the values are returned correctly

File: core/web/schema/type/job_run.graphql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+type JobRun {
```
Comment (jkongie) [inline core/web/schema/type/job_run.graphql:1]: This should have an id

**Review by jkongie — CHANGES_REQUESTED** [review]

File: core/web/schema/type/job_run.graphql (hunk 4 above)
Comment (vyzaldysanchez) [reply core/web/schema/type/job_run.graphql:1]: done

**Review by jkongie — APPROVED** [review]
Looks good

**Comment by vyzaldysanchez** [comment]
@cl-homu r+

**Comment by cl-homu** [comment]
:pushpin: Commit aa27f27d9c0e41a6a4ce87bfca2b8d74d8e68589 has been approved by `vyzaldysanchez`

**Comment by cl-homu** [comment]
:hourglass: Testing commit aa27f27d9c0e41a6a4ce87bfca2b8d74d8e68589 with merge 862e66e46415183cb0c8728a385433616bb587d4...

**Comment by cl-homu** [comment]
:sunny: Test successful - [checks-core](https://github.com/smartcontractkit/chainlink/runs/4240166566?check_suite_focus=true), [checks-lint](https://github.com/smartcontractkit/chainlink/runs/4240166820?check_suite_focus=true), [checks-npm](https://github.com/smartcontractkit/chainlink/runs/4240166902?check_suite_focus=true), [checks-operator](https://github.com/smartcontractkit/chainlink/runs/4240166562?check_suite_focus=true), [checks-prettier](https://github.com/smartcontractkit/chainlink/runs/4240167001?check_suite_focus=true), [checks-solidity](https://github.com/smartcontractkit/chainlink/runs/4240166660?check_suite_focus=true)
Approved by: vyzaldysanchez
Pushing 862e66e46415183cb0c8728a385433616bb587d4 to develop...

**Pull request merged**